### PR TITLE
some ruff C4 fixes in combinat

### PR DESCRIPTION
**Pull request by fchapoton**
this is fixing the `ruff` warnings for codes `C41*` in `combinat`

scripted using `ruff`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

**Comment by mantepse** [comment]
Looks good!  Let's just wait for the bots!

**Comment by fchapoton** [comment]
I have a doubt about one change in `affine_factorization`. Removing "reversed" is correct but may still be a wrong move.

**Comment by fchapoton** [comment]
hmm, at least one of these changes is breaking something

File: src/sage/combinat/affine_permutation.py [hunk 1]
```diff
@@ -2245,7 +2245,7 @@ def one(self):
             True
             sage: TestSuite(A).run()
         """
-        return self([i for i in range(1,self.k+2)])
+        return self(list(range(1,self.k+2)))
```
Comment (mkoeppe) [inline src/sage/combinat/affine_permutation.py]:
```suggestion
        return self(list(range(1, self.k + 2)))
```

File: src/sage/combinat/crystals/affine_factorization.py [hunk 2]
```diff
@@ -324,7 +324,7 @@ def bracketing(self, i):
                     right_n.remove(min(l))
                 else:
                     left_unbracketed += [m]
-            return [[j for j in left_unbracketed],[j for j in right_n]]
+            return [list(left_unbracketed),list(right_n)]
```
Comment (mkoeppe) [inline src/sage/combinat/crystals/affine_factorization.py]:
```suggestion
            return [list(left_unbracketed), list(right_n)]
```

File: src/sage/combinat/crystals/infinity_crystals.py [hunk 3]
```diff
@@ -251,7 +251,7 @@ def module_generator(self):
             [[1, 1, 1], [2, 2], [3]]
         """
         n = self._cartan_type.rank()
-        p = Partition([x for x in reversed(range(1, n+1))])
+        p = Partition(list(reversed(range(1, n+1))))
```
Comment (mkoeppe) [inline src/sage/combinat/crystals/infinity_crystals.py]:
```suggestion
        p = Partition(list(reversed(range(1, n + 1))))
```

File: src/sage/combinat/crystals/kirillov_reshetikhin.py [hunk 4]
```diff
@@ -3998,7 +3998,7 @@ def horizontal_dominoes_removed(r, s):
         sage: sage.combinat.crystals.kirillov_reshetikhin.horizontal_dominoes_removed(3,2)
         [[], [2], [2, 2], [2, 2, 2]]
     """
-    ulist = [ [y for y in x] + [0]*(r-x.length()) for x in partitions_in_box(r, s//2) ]
+    ulist = [ list(x) + [0]*(r-x.length()) for x in partitions_in_box(r, s//2) ]
```
Comment (mkoeppe) [inline src/sage/combinat/crystals/kirillov_reshetikhin.py]:
```suggestion
    ulist = [list(x) + [0]*(r-x.length()) for x in partitions_in_box(r, s//2)]
```

File: src/sage/combinat/derangements.py [hunk 5]
```diff
@@ -423,7 +423,7 @@ def cardinality(self):
             R = PolynomialRing(QQ, 'x', len(A))
             S = sum(R.gens())
             e = prod((S - x)**y for (x, y) in zip(R.gens(), A))
-            return Integer(e.coefficient(dict([(x, y) for (x, y) in zip(R.gens(), A)])))
+            return Integer(e.coefficient(dict(list(zip(R.gens(), A)))))
```
Comment (mkoeppe) [inline src/sage/combinat/derangements.py]:
```suggestion
            return Integer(e.coefficient(dict(zip(R.gens(), A))))
```

File: src/sage/combinat/designs/difference_family.py [hunk 6]
```diff
@@ -2937,7 +2937,7 @@ def complementary_difference_setsII(n, check=True):
     if t % 2 == 0:
         rho = G.multiplicative_generator()
         C0 = list({el**8 for el in G if el != 0})
-        C1, C2, C3, C6, C7 = map(lambda i: [rho**i * el for el in C0], [1, 2, 3, 6, 7])
+        C1, C2, C3, C6, C7 = ([rho**i * el for el in C0] for i in [1, 2, 3, 6, 7])
```
Comment (mkoeppe) [inline src/sage/combinat/designs/difference_family.py:2940]:
```suggestion
        C1, C2, C3, C6, C7 = [rho**i * el for el in C0] for i in [1, 2, 3, 6, 7]
```

File: src/sage/combinat/matrices/hadamard_matrix.py [hunk 7]
```diff
@@ -737,7 +737,7 @@ def construction_four_symbol_delta_code_I(X, Y, Z, W):
     def autocorrelation(seq, j):
         return sum([seq[i]*seq[i+j] for i in range(len(seq)-j)])
     for j in range(1, n):
-        assert sum(map(lambda seq: autocorrelation(seq, j), [X, Y, Z, W])) == 0
+        assert sum((autocorrelation(seq, j) for seq in [X, Y, Z, W])) == 0
```
Comment (mkoeppe) [inline src/sage/combinat/matrices/hadamard_matrix.py]:
```suggestion
        assert sum(autocorrelation(seq, j) for seq in [X, Y, Z, W]) == 0
```

File: src/sage/combinat/matrices/hadamard_matrix.py [hunk 8]
```diff
@@ -813,7 +813,7 @@ def autocorrelation(seq, j):
         return sum([seq[i]*seq[i+j] for i in range(len(seq)-j)])
 
     for j in range(1, n):
-        assert sum(map(lambda seq: autocorrelation(seq, j), [X, Y, Z, W])) == 0
+        assert sum((autocorrelation(seq, j) for seq in [X, Y, Z, W])) == 0
```
Comment (mkoeppe) [inline src/sage/combinat/matrices/hadamard_matrix.py]:
```suggestion
        assert sum(autocorrelation(seq, j) for seq in [X, Y, Z, W]) == 0
```

**Review by mkoeppe — APPROVED** [review]
LGTM.

**Pull request merged**